pkg/providers/v1/instances.go

```diff
@@ -195,22 +192,14 @@ type cacheCriteria struct {
 
 // describeAllInstancesCached returns all instances, using cached results if applicable
 func (c *instanceCache) describeAllInstancesCached(criteria cacheCriteria) (*allInstancesSnapshot, error) {
-	var err error
-	snapshot := c.getSnapshot()
-	if snapshot != nil && !snapshot.MeetsCriteria(criteria) {
-		snapshot = nil
-	}
-
-	if snapshot == nil {
-		snapshot, err = c.describeAllInstancesUncached()
-		if err != nil {
-			return nil, err
-		}
-	} else {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	if c.snapshot != nil && c.snapshot.MeetsCriteria(criteria) {
 		klog.V(6).Infof("EC2 DescribeInstances - using cached results")
+		return c.snapshot, nil
 	}
 
-	return snapshot, nil
+	return c.describeAllInstancesUncached()
 }
 
 // getSnapshot returns a snapshot if one exists
```

Comment:
Should we remove the `getSnapshot` method since its not being used anywhere else?